p2p.go

```diff
@@ -413,28 +381,28 @@ func (h *Host) AddUnicastPubSub(topic string, callback HandleUnicast) error {
 // AddBroadcastPubSub adds a broadcast topic that the host will pay attention to. This need to be called before using
 // Connect/JoinOverlay. Otherwise, pubsub may not be aware of the existing overlay topology
 func (h *Host) AddBroadcastPubSub(topic string, callback HandleBroadcast) error {
-	if _, ok := h.pubs[topic]; ok {
-		return nil
-	}
 	blacklist, err := NewLRUBlacklist(h.cfg.BlackListLRUSize)
 	if err != nil {
 		return err
 	}
 	pub, err := h.newPubSub(
 		h.ctx,
 		h.host,
-		pubsub.WithMessageSigning(true),
-		pubsub.WithStrictSignatureVerification(true),
 		pubsub.WithBlacklist(blacklist),
 	)
 	if err != nil {
 		return err
 	}
-	sub, err := pub.Subscribe(topic)
+
+	t, err := pub.Join(topic)
+	if err != nil {
+		return err
+	}
+
+	sub, err := t.Subscribe()
 	if err != nil {
 		return err
 	}
-	h.pubs[topic] = pub
```

Comment:
Is there a specific reason you've removed this? Do we not need to record the publisher?

Reply:
The pub sub stuff works a bit differently in the latest release. 

You publish on the topic itself so you don't need to store multiple publishers.

```
topics    map[string]*pubsub.Topic
```

And then to publish to that topic you do

```
	t, ok := h.topics[topic]
	if !ok {
		return nil
	}
	return t.Publish(context.Background(), data)
```

Reply:
ah, the same as subscribing. ok looks good!